### PR TITLE
A11y 9 hoc overview alt text

### PR DESCRIPTION
**Pull request by daynew**
a preview of the hourofcode overview page which doesn't have alt text

The images on https://code.org/hourofcode/overview, which are summaries of the courses, are images missing alt text. This PR adds alt text to the images.
* Puts the course upsells in alphabetical order so it's easier to find them.
* Add `image_alt` text attribute for the course upsells which are featured on the hourofcode.com/overview page. This text is added to the `alt` attribute of the `images` used to advertise a course.

## How to review this PR?
There are a lot of changes which are a actually just strings being moved around because I put the course upsells in alphabetical order. For that reason, I recommend you focus on the `image_alt` strings which have been added. Comment if you think they are an appropriate description of the image.

## What are the images
Here is a video of all the "tall" images which could potentially be view on this page:

https://user-images.githubusercontent.com/1372238/208177230-23dbb19c-c6d5-41c3-96ab-f884cc8dbb51.mov

## Links
* [JIRA](https://codedotorg.atlassian.net/browse/A11Y-9)

## Testing story
* Visited http://localhost.code.org:3000/hourofcode/overview
  * Inspected the HTML and confirmed all the images have the `alt` attribute
  * Ran Lighthouse in Chrome and confirmed that the tool is no longer reporting the images as missing alt text.

### CODE REVIEW

File: dashboard/config/locales/en.yml [hunk 1]
```diff
@@ -692,82 +692,156 @@ en:
       body_short: "Design an app, code with blocks or JavaScript to make it work, then share your app in seconds."
       audience: 'Ages 13+'
       cta: 'Learn how'
-    dance:
-      title: 'Dance Party'
+      image_alt: 'A purple bunny is wearing a rocket powered backpack.'
```
Comment (ebeastlake) [inline dashboard/config/locales/en.yml]: nit: Applies to many of these, do we care about active vs. passive voice for alt text? Ex. `A purple bunny wears a rocket-powered backpack.`

File: dashboard/config/locales/en.yml [hunk 2]
```diff
@@ -692,82 +692,156 @@ en:
...
-      cta: "Learn how"
-    frozen:
-      title: "Frozen"
-    starwars:
-      title: "Star Wars"
+      image_alt: 'A bar graph of letter frequencies in the English language.'
+    hoc:
+      title: 'Classic Maze'
+      body: 'Try the basics of computer science. Millions have given it a shot.'
+      image_alt: 'A collage of code, Mark Zuckerberg, and Angry Birds.'
```
Comment (ebeastlake) [inline dashboard/config/locales/en.yml]: nit: Should we try to make them all sentences? Eg. `A collage shows a code snippet, Mark Zuckerberg, and Angry Birds.`

File: dashboard/config/locales/en.yml [hunk 3]
```diff
@@ -692,82 +692,156 @@ en:
...
-      title: "Minecraft"
-      body: "Minecraft is back for the Hour of Code with a brand new activity! Journey through Minecraft with code."
+    oceans:
+      image_alt: 'A robot in the ocean scanning various sea creatures.'
+    playlab:
+      image_alt: 'A blue cat, goofy dog, and prehistoric weasel.'
+    poem-art:
+      title: 'Poem Art'
+      body: 'Express yourself! Use code to create animated poems.'
+      image_alt: 'A hipster squid performing on a stage.'
```
Comment (ebeastlake) [inline dashboard/config/locales/en.yml:802]: 😂 ❤️

Comment (daynew) [reply dashboard/config/locales/en.yml:802]: We need more fun images like this :)

**Comment by ebeastlake** [comment]
Something I would know if I understood all of this code better: Did you cover all of the course blocks so that `block_data` has an `image_alt` attribute for every possibility, or could we make it explicit that it falls back to an empty alt tag (or does it do that by default if nothing is found)?

**Comment by daynew** [comment]
> Something I would know if I understood all of this code better: Did you cover all of the course blocks so that `block_data` has an `image_alt` attribute for every possibility, or could we make it explicit that it falls back to an empty alt tag (or does it do that by default if nothing is found)?

Nice catch. I will investigate that and make sure it defaults to `""` if nothing is found.

**Comment by daynew** [comment]
> Something I would know if I understood all of this code better: Did you cover all of the course blocks so that `block_data` has an `image_alt` attribute for every possibility, or could we make it explicit that it falls back to an empty alt tag (or does it do that by default if nothing is found)?
I have addressed this issue here: https://github.com/code-dot-org/code-dot-org/pull/49589/commits/ad01ce20a8035ff56a42fa5c2b3b980239c425c6

This was a nice catch. If no string was found, it was actually going to produce the message "translations missing for string key..."

**Review by ebeastlake — APPROVED** [review]

**Pull request merged**